ts/docs.ts

```diff
@@ -48,49 +86,116 @@ async function docs(): Promise<void> {
         const schemaMdFile = path.join('public', `${title}.schema.md`)
         await encoda.write(articleMd, schemaMdFile)
 
-        const mdFile = path.join('schema', `${title}.md`)
-        if (await fs.pathExists(mdFile)) {
-          const article = await encoda.read(mdFile)
+        const maybeMdFile = path.join('schema', `${title}.md`)
+        const mdFileExists = await fs.pathExists(maybeMdFile)
+        const mdFile = mdFileExists ? maybeMdFile : schemaMdFile
+        const article = await encoda.read(mdFile)
+        // Don't output an HTML file if the Markdown document cannot be decoded into an article
+        if (!isArticle(article)) {
+          log.warn(
+            `docs.ts#57: cant decode '${mdFile}' as 'Article'; skipping HTML generation`
+          )
+          return
+        }
 
-          // Don't output an HTML file if the Markdown document cannot be decoded into an article
-          if (!isArticle(article)) {
-            return
-          }
+        log.debug(`schema2Article(${jsonFile}) => ${schemaMdFile}`)
+        if (!mdFileExists) {
+          stats.htmlFromYamls++
+          log.info(
+            `(${stats.htmlFromYamls})${maybeMdFile} doc not found; stubbing html from schema2Article(${mdFile})`
+          )
+        }
 
-          const processed = await encodaProcess(
+        // dom fragment to let readers know this file was generated automatically
+        const subheadingContent = !mdFileExists
+          ? [
+              {
+                type: 'Heading',
+                depth: 6,
+                content: [
+                  `note: this html file was autogenerated by 'ts/docs.ts::schema2Article()' from '${schemaMdFile}'; create documentation for it at '${maybeMdFile}'`
+                ],
+                id: 'notice'
+              }
+            ]
+          : []
+
+        // can we find any examples to include?
+        // note: path-matching in getExamples() that looks for .yaml examples
+        // is not too reliable b/c I am not sure what the naming pattern is
+        const yamlExamples = await getExamples(title)
+        let exampleContent = yamlExamples.map((example: any) => {
+          return {
+            type: 'CodeBlock',
+            programmingLanguage: 'yaml',
+            text: example
+          }
+        })
+        if (exampleContent.length > 0) {
+          exampleContent = [
```

Comment:
I think we can merge these two blocks to to simplify the logic here by removing the `if` statement. Meaning we set the Headings inside the `yamlExamples.map(` statement.


Also to continue type casting changes, line `127` should be:
```diff
- let exampleContent = yamlExamples.map((example: any) => {
+ let exampleContent = yamlExamples.map((example: string) => {
```